src/modules/map/components/contextMenu/MapContextMenu.js

```diff
@@ -22,77 +21,66 @@ export class MapContextMenu extends BaElement {
 		this._translationService = translastionService;
 	}
 
+	_calculateSector(coordinate) {
 
-	_calculateParameter(coordinate, element) {
-		//alignment: location of the event coordinate relative to the menu
-		const values = { left: 0, top: 0, vAlignment: 'left', hAlignment: 'top' };
-		if (!element || !coordinate) {
-			return values;
-		}
+		const widthBorder = window.innerWidth * .66;
+		const heightBorder = window.innerHeight * .66;
 
-		//see: css classes
-		const vOffset = 20;
-		const offsetBorderInPercent = 0.2;
-		const menuWidth = element.offsetWidth;
-		const menuHeight = element.offsetHeight + vOffset;
-		const windowWidth = window.innerWidth - (window.innerWidth * offsetBorderInPercent);
-		const windowHeight = window.innerHeight - (window.innerHeight * offsetBorderInPercent);
-
-		if ((windowWidth - coordinate[0]) < menuWidth) {
-			values.vAlignment = 'right';
-			values.left = coordinate[0] - menuWidth + 'px';
-		}
-		else {
-			values.vAlignment = 'left';
-			values.left = coordinate[0] + 'px';
+		//windiw sector the click event occurred:
+		//0-1
+		//3-2	
+
+		if (coordinate[0] <= widthBorder && coordinate[1] <= heightBorder) {
+			return 0;
 		}
-		if ((windowHeight - coordinate[1]) < menuHeight) {
-			values.hAlignment = 'bottom';
-			values.top = coordinate[1] - menuHeight + 'px';
+		else if (coordinate[0] > widthBorder && coordinate[1] <= heightBorder) {
+			return 1;
 		}
-		else {
-			values.hAlignment = 'top';
-			values.top = coordinate[1] + vOffset + 'px';
+		else if (coordinate[0] > widthBorder && coordinate[1] > heightBorder) {
+			return 2;
 		}
-		return values;
+		return 3;
```

Comment:
Do we discussed  already named_values for magic numbers like 0-3? For instance `const VIEW_QUADRANT_UPPER_LEFT =0` etc.

Reply:
Good point, but let's wait until we can use static fields. We want to use such named values also outside this single method. From my point of view it's confusing, if we would use member variables for that case, they should represent state...